components/token-asset/TokenHistoryItem.tsx

```diff
@@ -0,0 +1,84 @@
+import kasIcon from "@/assets/images/kas-icon.svg";
+import { formatUSD } from "@/lib/utils.ts";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { Op, TickerInfo } from "@/hooks/useKasplex.ts";
+import { TokenMetadata } from "@/hooks/useKasFyi.ts";
+import { twMerge } from "tailwind-merge";
+import { applyDecimal } from "@/lib/krc20.ts";
+
+type TokenHistoryItemProps = { op: Op; tickerInfo?: TickerInfo | undefined };
+
+export default function TokenHistoryItem({
+  op,
+  tickerInfo,
+}: TokenHistoryItemProps) {
+  const { ticker } = useParams();
+  const { fetchTokenMetadataByTicker } = useKasFyi();
```

Comment:
![image](https://github.com/user-attachments/assets/1863438d-c2e0-4716-a0da-36246fd6b7d2)
I would make sure `fetchTokenInfo` be called once when going into token asset page. Maybe it would be better to have parent component to handle the cache and refresh.

Reply:
23c79e872e35c519e6cc2b06e5921faf00be7d79
I used SWR for that.